src/query/expression/src/types/decimal.rs

```diff
@@ -687,19 +698,48 @@ impl Decimal for i256 {
 
     fn do_round_mul(self, rhs: Self, shift_scale: u32) -> Option<Self> {
         let div = i256::e(shift_scale);
-        if self.is_negative() == rhs.is_negative() {
+        let ret: Option<i256> = if self.is_negative() == rhs.is_negative() {
             self.checked_mul(rhs).map(|x| (x + div / 2) / div)
         } else {
             self.checked_mul(rhs).map(|x| (x - div / 2) / div)
-        }
+        };
+
+        ret.or_else(|| {
+            let a = BigInt::from_le_bytes(&self.to_le_bytes());
+            let b = BigInt::from_le_bytes(&rhs.to_le_bytes());
+            let div = BigInt::from(10).pow(shift_scale);
+            if self.is_negative() == rhs.is_negative() {
+                Self::from_bigint((a * b + div.clone() / 2) / div)
+            } else {
+                Self::from_bigint((a * b - div.clone() / 2) / div)
+            }
+        })
     }
 
-    fn do_round_div(self, rhs: Self, mul: Self) -> Option<Self> {
-        if self.is_negative() == rhs.is_negative() {
+    fn do_round_div(self, rhs: Self, mul_scale: u32) -> Option<Self> {
+        let fallback = || {
+            let a = BigInt::from_le_bytes(&self.to_le_bytes());
+            let b = BigInt::from_le_bytes(&rhs.to_le_bytes());
+            let mul = BigInt::from(10).pow(mul_scale);
```

Comment:
can be pre computed(maybe later)？the pow is implemented by multi mul_scale times, which is slow?

Reply:
fallback is already in a lambda function, so it's lazy computed.